File: sdks/typescript/src/opik/rest_api/serialization/types/BreakdownConfig.ts
// 此文件由 Fern 根据我们的 API 定义自动生成。

import type * as OpikApi from "../../api/index.js";
import * as core from "../../core/index.js";
import type * as serializers from "../index.js";
import { BreakdownConfigField } from "./BreakdownConfigField.js";

export const BreakdownConfig: core.serialization.ObjectSchema<
    serializers.BreakdownConfig.Raw,
    OpikApi.BreakdownConfig
> = core.serialization.object({
    field: BreakdownConfigField.optional(),
    metadataKey: core.serialization.property("metadata_key", core.serialization.string().optional()),
    subMetric: core.serialization.property("sub_metric", core.serialization.string().optional()),
});

export declare namespace BreakdownConfig {
    export interface Raw {
        field?: BreakdownConfigField.Raw | null;
        metadata_key?: string | null;
        sub_metric?: string | null;
    }
}

File: sdks/python/src/opik/rest_api/types/breakdown_config.py
# 此文件由 Fern 根据我们的 API 定义自动生成。

import typing

import pydantic
from ..core.pydantic_utilities import IS_PYDANTIC_V2, UniversalBaseModel
from .breakdown_config_field import BreakdownConfigField


class BreakdownConfig(UniversalBaseModel):
    field: typing.Optional[BreakdownConfigField] = None
    metadata_key: typing.Optional[str] = None
    sub_metric: typing.Optional[str] = None

    if IS_PYDANTIC_V2:
        model_config: typing.ClassVar[pydantic.ConfigDict] = pydantic.ConfigDict(extra="allow", frozen=True)  # type: ignore # Pydantic v2
    else:

        class Config:
            frozen = True
            smart_union = True
            extra = pydantic.Extra.allow
